### ISSUE TITLE
The focus outline of the Key Metrics Selection Panel's close button is slightly misaligned.

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#7498]
## Feature Description

When Key Metrics Selection Panel's close button is focused, the X should be vertically centered in the focus outline. It's currently marginally off center.

![image](https://github.com/google/site-kit-wp/assets/18395600/84bc2ed0-b731-4c80-82be-1e01f2c7f1f7)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* When Key Metrics Selection Panel's close button is focused, the X should be vertically centered in the focus outline.

## Implementation Brief

* adding `display: flex;` to the close button selector will fix the issue https://github.com/google/site-kit-wp/blob/b03c75c0af51101c95da85d5fdb8182b32c8713b/assets/sass/components/key-metrics/_googlesitekit-km-selection-panel.scss#L47

### Test Coverage

* Update VRTs/snapshot tests as-needed

## QA Brief

* Enable g4reporting and userInput flags
* Setup analytics and key metrics
* In the key metrics widget, click on Change Metrics button
* Click on close icon in the selection panel and hold for few seconds (or set :focus state on it from developer tools)
* Focus outline will appear - verify circle is properly rounded and X icon is not misaligned

## Changelog entry

* N/A

**Comment by tofumatt** [comment]
IB ✅

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- When Key Metrics Selection Panel's close button is focused, the X is vertically centered in the focus outline.
- Tested this on desktop, mobile, tablet and other small viewports to make sure the behaivour was consistent. 

<details>
 <summary>Scrennshots</summary>

![image](https://github.com/google/site-kit-wp/assets/73545194/1630eab3-2e7a-49ba-97fc-c44650597e8e)
![image](https://github.com/google/site-kit-wp/assets/73545194/636f682b-543c-406c-9a68-f7495fc0307b)

</details>
